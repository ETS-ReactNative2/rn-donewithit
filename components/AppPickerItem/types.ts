import { GestureResponderEvent } from 'react-native'

export type AppPickerItemProps = {
  label: string
  selected?: boolean
  onPress?: (e?: GestureResponderEvent) => void
  data?: any
}
