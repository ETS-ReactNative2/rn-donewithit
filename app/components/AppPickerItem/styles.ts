import { StyleSheet } from 'react-native'
import CONSTS from './consts'
import COLORS from '../../config/colors'

const styles = StyleSheet.create({
  text: {
    lineHeight: 50,
    paddingHorizontal: 20,
    fontSize: CONSTS.TEXT_INPUT_FONT_SIZE
  },
  selected: {
    color: COLORS.PRIMARY
  }
})

export default styles
