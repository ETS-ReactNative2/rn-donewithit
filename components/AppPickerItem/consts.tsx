const CONSTS = {
  TEXT_INPUT_FONT_SIZE: 18
}

export default CONSTS
