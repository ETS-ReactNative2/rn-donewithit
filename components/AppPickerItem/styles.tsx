import { StyleSheet } from 'react-native'
import CONSTS from './consts'

const styles = StyleSheet.create({
  text: {
    textAlign: 'center',
    lineHeight: 50,
    fontSize: CONSTS.TEXT_INPUT_FONT_SIZE
  }
})

export default styles
