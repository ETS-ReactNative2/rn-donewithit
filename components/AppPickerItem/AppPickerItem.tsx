import React, { useCallback } from 'react'
import { GestureResponderEvent, TouchableOpacity } from 'react-native'
import AppText from '../AppText'
import styles from './styles'
import { AppPickerItemProps } from './types'

function AppPickerItem({ label, onPress, selected }: AppPickerItemProps) {
  const handlePress = useCallback((e?: GestureResponderEvent) => {
    typeof onPress === 'function' && onPress(e)
  }, [])
  return (
    <TouchableOpacity onPress={handlePress}>
      <AppText style={[styles.text, selected && styles.selected]}>
        {label}
      </AppText>
    </TouchableOpacity>
  )
}

export default AppPickerItem
