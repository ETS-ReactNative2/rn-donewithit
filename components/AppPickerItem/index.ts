import AppPickerItem from './AppPickerItem'
export default AppPickerItem
export { AppPickerItemProps } from './types'
