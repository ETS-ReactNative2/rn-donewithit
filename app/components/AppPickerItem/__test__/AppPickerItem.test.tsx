import React from 'react'
import renderer from 'react-test-renderer'
import AppPickerItem from '../index'
import COLORS from '../../../config/colors'

jest.mock(
  '@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js',
  () => {
    return () => ''
  }
)

const option = {
  label: 'Furniture',
  iconName: 'lamp',
  iconBackgroundColor: COLORS.PRIMARY,
  value: 1
}

const props = {
  label: option.label,
  selected: true,
  data: option
}

describe('AppPickerItem', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppPickerItem {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
