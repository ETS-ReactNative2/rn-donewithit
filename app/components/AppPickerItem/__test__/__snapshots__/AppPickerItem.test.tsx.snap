// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPickerItem renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
          "fontFamily": "Courier",
          "fontSize": 13,
        },
        Array [
          Object {
            "fontSize": 18,
            "lineHeight": 50,
            "paddingHorizontal": 20,
          },
          Object {
            "color": "tomato",
          },
        ],
      ]
    }
  >
    Furniture
  </Text>
</View>
`;
